libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -471,6 +492,180 @@ uint64_t VoteManager::roundDeterminedFromVotes(size_t two_t_plus_one) {
   return 0;
 }
 
+void VoteManager::sendRewardPeriodCertVotes(uint64_t reward_period) {
+  auto reward_period_cert_votes = db_->getCertVotes(reward_period);
+  if (reward_period_cert_votes.empty()) {
+    return;
+  }
+
+  auto net = network_.lock();
+  assert(net);
+  net->onNewPbftVotes(std::move(reward_period_cert_votes), true);
+}
+
+bool VoteManager::addRewardVote(const std::shared_ptr<Vote>& vote) {
+  // Don't check the vote if in DB here. That will check in updateRewardVotes for reducing DB read
+  const auto vote_hash = vote->getHash();
+  std::unique_lock lock(reward_votes_mutex_);
+  if (reward_votes_.second.contains(vote_hash)) {
+    return false;
+  }
+
+  const auto voted_block_hash = vote->getBlockHash();
+  const auto pbft_chain_last_block_hash = pbft_chain_->getLastPbftBlockHash();
+  if (voted_block_hash != pbft_chain_last_block_hash) {
+    LOG(log_dg_) << "Drop reward vote " << vote_hash << ". PBFT chain last block is " << pbft_chain_last_block_hash
+                 << ", but vote for PBFT block is " << voted_block_hash;
+    return false;
+  }
+
+  if (!reward_votes_.first) {
+    reward_votes_.first = voted_block_hash;
+  } else if (reward_votes_.first != pbft_chain_last_block_hash) {
+    // Clear missing reward votes
+    LOG(log_dg_) << "Remove reward votes: \n";
+    for (const auto& v : reward_votes_.second) {
+      LOG(log_dg_) << v.first;
```

Comment:
dont log each hash on separatre line, just create string with all hashes divided by whitespace and use 1 log

Reply:
fixed